packages/xchain-cosmos/src/client.ts

```diff
@@ -254,7 +254,10 @@ class Client implements CosmosClient, XChainClient {
             amount: baseAmount(balance.amount, DECIMAL),
           }
         })
-        .filter((balance) => !asset || assetToString(balance.asset) === assetToString(asset))
+        .filter(
+          (balance) =>
+            !assets || assets.findIndex((asset) => assetToString(balance.asset) === assetToString(asset)) >= 0,
```

Comment:
As same as before: `findIndex`[ is not supported in IE](https://caniuse.com/?search=findIndex).